tests/prefetch_related/tests.py

```diff
@@ -291,6 +292,15 @@ def test_m2m_then_reverse_one_to_one_object_ids(self):
         sql = queries[-1]['sql']
         self.assertWhereContains(sql, self.author1.id)
 
+    def test_filter_deferred(self):
+        add_q = Query.add_q
+        with mock.patch.object(
+            Query, 'add_q', autospec=True,
+            side_effect=lambda self, q: add_q.__get__(self, Query)(q),
+        ) as mocked_add_q:
+            list(Book.objects.prefetch_related('authors'))
+            self.assertEqual(mocked_add_q.call_count, 1)
+
```

Comment:
I was about to suggest we also test that related querysets are actually filtered by this is already covered by the tests added in c92123cc1dceeb800b3b8900e2e530ed19d78863#diff-dc510a0902e56d1afd660af76571c9ef.